x-pack/plugins/apm/public/hooks/use_service_name.tsx

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useHistory, matchPath } from 'react-router-dom';
+import { apmRouteConfig } from '../components/routing/apm_route_config';
+
+export function useServiceName(): string | undefined {
+  const history = useHistory();
+  for (const config of apmRouteConfig) {
+    const match = matchPath<{ serviceName?: string }>(
```

Comment:
Would it be better to use [`useRouteMatch`](https://github.com/ReactTraining/react-router/blob/master/packages/react-router/docs/api/hooks.md#useroutematch)? I don't think we would need to iterate through the routes if we did that. It's independent of any route component and just uses the current path.

Reply:
`useRouteMatch()` returns an empty match object:

![image](https://user-images.githubusercontent.com/352732/124237595-fb31ae80-db17-11eb-984b-14ac03ccbec8.png)

Reply:
Don't you have to give it a path like `useRouteMatch('/app/apm/services/:serviceName/*')`? I thought that's how it worked.

Reply:
I think in this case we only know the actual pathname (`history.location.pathname`), not the route since it's a generic hook that can be called anywhere in the application.

Reply:
... but seeing your example, if it allows wildcards it might be different.

Reply:
nvm. It seems like `useRouteMatch` simply uses https://github.com/pillarjs/path-to-regexp under the hood. What @smith is suggesting makes sense

Reply:
I've changed it to `useRouteMatch('/services/:serviceName')`, which seems to work fine.